eventstore/store.go

```diff
@@ -176,7 +184,8 @@ func (s *Store) Threadservice() threadservice.Threadservice {
 func (s *Store) Register(name string, defaultInstance interface{}) (*Model, error) {
 	s.lock.Lock()
 	defer s.lock.Unlock()
-	if s.alreadyRegistered(defaultInstance) {
+
+	if _, ok := s.modelNames[name]; ok {
```

Comment:
I now added a set-map of model registered `Name`s, so we use that for checking if the model is already registered. Before, the GoTyped map was used... but in JSON Mode no Go Types are registered so that map will be always empty.

I guess this would be the right thing to do in advance: use the `Name` that the client chose for the Model as their identity. (The Go Type isn't really an identity, just a representation).